hkmc2/shared/src/main/scala/hkmc2/syntax/Tree.scala

```diff
@@ -157,10 +157,9 @@ object PlainTup:
   def apply(fields: Tree*): Tree = Tup(fields.toList)
 
 object Apps:
-  def unapply(t: Tree): Opt[(Ident, Ls[Tup])] = t match
+  def unapply(t: Tree): Opt[(Tree, Ls[Tup])] = t match
```

Comment:
If this always matches, you should change the return type to `S[(Tree, Ls[Tup])]`.